src/pydargs/__init__.py

```diff
@@ -85,14 +111,14 @@ def _add_arguments(parser: ArgumentParser, tp: Type[Dataclass], prefix: str = ""
             if len(argument_kwargs["help"]):
                 argument_kwargs["help"] += " "
             argument_kwargs["help"] += f"(default: {field.default})"
-        if "metavar" in field.metadata:
-            argument_kwargs["metavar"] = field.metadata["metavar"]
         positional = field.metadata.get("positional", False)
         short_option = field.metadata.get("short_option")
         if positional:
+            if has_subparser:
+                warn("Positional arguments defined after a subparser cannot be parsed.")
```

Comment:
What is the behavior when this happens? Will this raise the error that some part of the namespace is unused? if so, should we raise here immediately?

Reply:
The behaviour is that is would expect `entrypoint ACTION ... [positional]`, but that the `positional` is always parsed by the subparser connected to the action. If the positional has a default, then this is not necessarily an issue (but you can't configure the field with an argument). If it doesn't, the parse can't be used. But this will always yield an argparse error that is fairly legible.

Reply:
Yeah okay so this is related to the global parameter thing in the discussion below.. I see 👍